throw error if no pipeline

jimsheldon opened the pull request.

krishna-koushik committed:
throw error if no pipeline

jimsheldon commented:
This should fix #14

microadam commented:
Hi @jimsheldon. Thanks for the contribution. I am not sure this will work though, as this seems to assume that there will only be ONE yaml doc in the drone file. There could be multiple pipelines or even things like:

```
---
kind: secret
```

This PR assumes that it will only contain ONE yaml doc, so if you used a `secret` doc, it would break as you are throwing within a `map` function

jimsheldon commented:
Thanks @microadam we will look into this

Commit:
check if 'kind' is undefined

if 'kind' is undefined, the config could be 0.8 syntax, so return a 204 and let
the drone server retrieve it

if 'kind' is anything other than 'pipeline', return it

jimsheldon commented:
@microadam I pushed a fix that should handle non-pipeline kinds properly

microadam commented:
Thanks for this! I will try and have a proper look through this soon. I really need to get some unit tests on this project so we can ensure no existing functionality breaks with little changes like this.
